optuna/multi_objective/hypervolume/exact_2d.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+
+from optuna.multi_objective.hypervolume import BaseHypervolume
+from optuna.multi_objective.hypervolume import compute_2points_volume
+
+
+class Exact2d(BaseHypervolume):
+    """Hypervolume calculator in the 2-dimensional case.
+
+    This class exactly calculates the hypervolume in the 2-dimensional case.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def compute(self, solution_set: np.ndarray, reference_point: np.ndarray) -> float:
```

Comment:
Just for the record, tried with creating a tiled array + masking but it was a lot slower than your looped version. Likely because of the n^2 memory allocation. For those who are interested.
```python
        solution_set = solution_set[solution_set[:, 0].argsort()]

        n = solution_set.shape[0]
        widths = reference_point[1] - solution_set[:, 1]
        widths_masked = np.repeat(widths[:, None], n, axis=1)
        widths_masked = widths_masked.astype(np.float32)
        widths_masked[np.tril_indices(n, k=-1)] = -np.inf
        widths = widths_masked.max(axis=0)

        heights =  np.append(solution_set[1:, 0], reference_point[0]) - solution_set[:, 0]

        return np.sum(widths * heights)
```

Reply:
Thanks for the investigation. The logic of the function is left as it is, but I've modified the naming conventions for the variables based on you.